base/mpfr.jl

```diff
@@ -571,9 +572,8 @@ end
 
 for f in (:log, :log2, :log10)
     @eval function $f(x::BigFloat)
-        if x < 0
-            throw(DomainError(x, string($f, " will only return a complex result if called ",
-                              "with a complex argument. Try ", $f, "(complex(x)).")))
+        if x < 0 && throw(DomainError(x, string($f, " will only return a complex result if called ",
```

Comment:
I don't understand, the one linked uses `if x < -1`, i.e. a normal `if` block? Also did you mean to remove the `if` here? otherwise this form `if cond && throw(...) end` is very strange.